failed to start language server "next-ls" error

### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

I have the following settings
```
{
  "elixir": {
    "lsp": "next_ls"
  }
}
```
When I run zed, it fails to start the language server

I've checked and it seems that zed couldn't download(or extract) `next-ls` because this directory is empty : 

```
ls -l -d "/Users/user/Library/Application Support/Zed/languages/next-ls"
drwxr-xr-x  3 user  staff  96 Oct 27 10:19 /Users/milad/Library/Application Support/Zed/languages/next-ls/
```

### Environment

Zed: v0.109.2 (stable)
OS: macOS 14.0.0
Memory: 32 GiB
Architecture: x86_64

### If applicable, add mockups / screenshots to help explain present your vision of the feature

_No response_

### If applicable, attach your `~/Library/Logs/Zed/Zed.log` file to this issue.
If you only need the most recent lines, you can run the `zed: open log` command palette action to see the last 1000.


```
2023-10-27T06:20:09 [INFO] 0 unhandled notification LogMessage:
{
  "level": 0,
  "message": "[DEBUG] [agent] [2023-10-27T06:20:09.040Z] Telemetry initialized",
  "metadataStr": "[DEBUG] [agent] [2023-10-27T06:20:09.040Z]",
  "extra": [
    "Telemetry initialized"
  ]
}
2023-10-27T06:20:12 [ERROR] oneshot canceled
2023-10-27T08:13:10 [INFO] ========== starting zed ==========
2023-10-27T08:13:10 [INFO] Opening main db
2023-10-27T08:13:10 [INFO] set environment variables from shell:/usr/local/bin/fish, path:/usr/local/opt/openjdk/bin:/Users/user/.asdf/shims/:/Users/user/.cargo/bin:/usr/local/bin:/Users/user/.cargo/bin:/System/Cryptexes/App/usr/bin:/Users/user/.cargo/bin:/usr/bin:/Users/user/.cargo/bin:/bin:/Users/user/.cargo/bin:/usr/sbin:/Users/user/.cargo/bin:/sbin:/Users/user/.cargo/bin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/local/bin:/Users/user/.cargo/bin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/bin:/Users/user/.cargo/bin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/appleinternal/bin:/Users/user/.cargo/bin:/Library/Apple/usr/bin:/Users/user/.cargo/bin:/usr/local/share/dotnet:/Users/user/.cargo/bin:~/.dotnet/tools:/Users/user/.cargo/bin:/Users/user/Library/Application Support/Coursier/bin
2023-10-27T08:13:10 [INFO] Opening main db
2023-10-27T08:13:10 [INFO] build git repository ".git"
2023-10-27T08:13:12 [INFO] Node runtime install_if_needed
2023-10-27T08:13:12 [INFO] Opening main db
2023-10-27T08:13:13 [INFO] starting language server "next-ls", path: "/Users/user/dev/my-project", id: 1
2023-10-27T08:13:14 [ERROR] failed to start language server "next-ls": Bad CPU type in executable (os error 86)
2023-10-27T08:13:14 [INFO] About to spawn test binary
2023-10-27T08:13:14 [WARN] test binary failed to launch
2023-10-27T08:13:14 [WARN] test binary check failed
2023-10-27T08:13:14 [INFO] beginning to reinstall server
2023-10-27T08:13:14 [INFO] deleting server container
2023-10-27T08:13:14 [INFO] starting language server "next-ls", path: "/Users/user/dev/my-project", id: 2
2023-10-27T08:13:16 [ERROR] failed to start language server "next-ls": Bad CPU type in executable (os error 86)
2023-10-27T08:13:16 [INFO] About to spawn test binary
2023-10-27T08:13:16 [WARN] test binary failed to launch
2023-10-27T08:13:16 [WARN] test binary check failed
2023-10-27T08:13:16 [INFO] beginning to reinstall server
2023-10-27T08:13:16 [INFO] deleting server container
2023-10-27T08:13:16 [INFO] starting language server "next-ls", path: "/Users/user/dev/my-project", id: 3
2023-10-27T08:13:16 [ERROR] failed to start language server "next-ls": Bad CPU type in executable (os error 86)
2023-10-27T08:13:16 [INFO] About to spawn test binary
2023-10-27T08:13:16 [WARN] test binary failed to launch
2023-10-27T08:13:16 [WARN] test binary check failed
2023-10-27T08:13:16 [INFO] beginning to reinstall server
2023-10-27T08:13:16 [INFO] deleting server container
2023-10-27T08:13:17 [INFO] starting language server "next-ls", path: "/Users/user/dev/my-project", id: 4
2023-10-27T08:13:17 [INFO] 0 unhandled notification LogMessage:
```

--- a/crates/zed/src/languages/elixir.rs
+++ b/crates/zed/src/languages/elixir.rs
@@ -321,8 +321,8 @@ impl LspAdapter for NextLspAdapter {
             latest_github_release("elixir-tools/next-ls", false, delegate.http_client()).await?;
         let version = release.name.clone();
         let platform = match consts::ARCH {
-            "x86_64" => "darwin_arm64",
-            "aarch64" => "darwin_amd64",
+            "x86_64" => "darwin_amd64",
+            "aarch64" => "darwin_arm64",
             other => bail!("Running on unsupported platform: {other}"),
         };
         let asset_name = format!("next_ls_{}", platform);

